src/objects/embed/embed.scss

```diff
@@ -0,0 +1,119 @@
+@use 'sass:meta';
+@use '../../design-tokens/aspect-ratio.yml';
+@use '../../design-tokens/breakpoint.yml';
+
+/**
+ * We define the default ratio up-front to avoid outputting unnecessary modifier
+ * classes for it.
+ */
+
+$default-aspect-ratio: aspect-ratio.$square;
+
+/* stylelint-disable suitcss/custom-property-no-outside-root */
+
+/**
+ * Makes ratio modifier classes based on the `aspect-ratio` and `breakpoint`
+ * design tokens. Intended to be included from within the base class.
+
+ * We iterate over breakpoints before ratios so wider breakpoints will be later
+ * in the cascade. If we iterated over aspect ratios first, then a narrower
+ * breakpoint would override a wider one if its aspect ratio came first in our
+ * design token file.
+ *
+ * 1. For some reason Sass does not compile `$ratio` without this string
+ *    interpolation syntax. I'm not entirely sure why.
+ * 2. It's not super DRY to repeat this line twice in the mixin, but every
+ *    alternative I explored took a lot more code.
+ */
+
+@mixin aspect-ratio-modifiers() {
+  @each $name, $ratio in meta.module-variables('aspect-ratio') {
+    @if $ratio != $default-aspect-ratio {
+      &--#{$name} {
+        --aspect-ratio: #{$ratio}; /* 1 */
+      }
+    }
+  }
+  @each $suffix, $breakpoint in meta.module-variables('breakpoint') {
+    @media (min-width: $breakpoint) {
+      @each $name, $ratio in meta.module-variables('aspect-ratio') {
+        &--#{$name}\@#{$suffix} {
+          --aspect-ratio: #{$ratio}; /* 2 */
+        }
+      }
+    }
+  }
+}
+
+/**
+ * This pattern uses the now-classic padding box hack, with a modern twist. The
+ * aspect ratio is defined via a custom property, which allows us arbitrary
+ * ratios based on content needs.
+ *
+ * @see https://css-tricks.com/aspect-ratio-boxes/#article-header-id-7
+ */
+
+/**
+ * Fallback behavior: Browsers that do not support custom properties will use
+ * the element's intrinsic aspect ratio. These styles cause the element to fill
+ * the available width.
+ */
+
+.o-embed > * {
+  display: block;
+  width: 100%;
+}
```

Comment:
Question: Down below we target `picture > img` specially. Do we need to do something similar in our fallback? Though I guess IE11 doesn't support `picture` anyways...

Reply:
Great question!